indigo/indigo/src/main/scala/indigo/platform/audio/AudioPlayer.scala

```diff
@@ -113,9 +114,29 @@ final class AudioPlayer(context: AudioContextProxy):
   private def findAudioDataByName(assetName: AssetName): Option[dom.AudioBuffer] =
     soundAssets.find(a => a.name == assetName).map(_.data)
 
-  def playSound(assetName: AssetName, volume: Volume): Unit =
+  private val soundNodes: mutable.Map[AssetName, AudioNodes] = mutable.HashMap.empty
+
+  @SuppressWarnings(Array("scalafix:DisableSyntax.null"))
+  private def stopSound(assetName: AssetName): Unit =
+    soundNodes.remove(assetName).foreach { source =>
+      source.audioBufferSourceNode.onended = null
+      source.audioBufferSourceNode.stop()
+    }
+
+  private def stopAllSound(): Unit =
+    soundNodes.keySet.foreach(stopSound)
+
+  def playSound(assetName: AssetName, volume: Volume, switch: SoundSwitch): Unit =
     findAudioDataByName(assetName).foreach { sound =>
-      setupNodes(sound, volume, loop = false).audioBufferSourceNode.start(0)
+      val node = setupNodes(sound, volume, loop = false)
+      node.audioBufferSourceNode.onended = _ => soundNodes.remove(assetName)
+      node.audioBufferSourceNode.start(0)
+      switch match {
+        case SoundSwitch.StopAll          => stopAllSound()
+        case SoundSwitch.StopPreviousSame => stopSound(assetName)
+        case SoundSwitch.Continue         => // noop
```

Comment:
Can this be `=> ()` please? It's probably equivalent, but I prefer something explicit.

Reply:
Addressed in https://github.com/PurpleKingdomGames/indigo/pull/877/commits/5f02245f246299278ac5fc7a6e196e71d9713dff